cmake/git.cmake

```diff
@@ -24,9 +24,9 @@ if(NOT BUILD_VERSION AND GIT_FOUND)
     RESULT_VARIABLE                  return_code
   )
   if(return_code)
-    message(STATUS "git describe failed: ${stderr};
-   BUILD_VERSION is left undefined")
-  elseif(CMAKE_VERBOSE_MAKEFILE)
-    message(STATUS "git describe stderr: ${stderr}")
+    message(STATUS "git describe failed: ${stderr}")
+  elseif(NOT "${stderr}" STREQUAL "")
+    message(STATUS "git describe warned: ${stderr}")
   endif()
+  message(STATUS "BUILD_VERSION=${BUILD_VERSION}")
```

Comment:
when this commit was merged https://github.com/zephyrproject-rtos/zephyr/commit/a880fb1343261a328bf2939cfdf2d19ef7447408 the dependency on `version.cmake` was removed:
https://github.com/zephyrproject-rtos/zephyr/blob/01c1bcf72489b363adaefb41db7e5cd746ea197d/cmake/app/boilerplate.cmake#L158

This means that we can actually move `git.cmake` before `version.cmake` like this:
```
include(${ZEPHYR_BASE}/cmake/git.cmake)
include(${ZEPHYR_BASE}/cmake/version.cmake)  # depends on hex.cmake
```

and then print all version information together, for example like this:
```
-- Application: /projects/github/ncs/zephyr/samples/hello_world
-- Zephyr version: 2.5.99 (/projects/github/ncs/zephyr) git=zephyr-v2.5.0-2958-g704302800aeb
```

Instead of scattered like this:
```
-- Application: /projects/github/ncs/zephyr/samples/hello_world
-- Zephyr version: 2.5.99 (/projects/github/ncs/zephyr)
-- Found Python3: /usr/bin/python3.8 (found suitable exact version "3.8.5") found components: Interpreter 
-- Found west (found suitable version "0.10.99", minimum required is "0.7.1")
-- BUILD_VERSION=zephyr-v2.5.0-2958-g704302800aeb
```

I think that would be a nice improvement for this PR.

Reply:
Thanks @tejlmand. I agree it would look better but massively better. It would also be a slightly bigger PR with one additional and independent change (likely in a separate commit) and I'm reluctant to request another round of review from all the people who already approved :-/

> -- Zephyr version: 2.5.99 (/projects/github/ncs/zephyr) git=zephyr-v2.5.0-2958-g704302800aeb

BUILD_VERSION can be overridden and not come from git.

> when this commit was merged a880fb1 the dependency on version.cmake was removed:

Did you mean another commit? I don't see any logical change in a880fb1

Reply:
>  I'm reluctant to request another round of review from all the people who already approved :-/

I would +1

Reply:
> >    `-- Zephyr version: 2.5.99 (/projects/github/ncs/zephyr) git=zephyr-v2.5.0-2958-g704302800aeb`
>
>  BUILD_VERSION can be overridden and not come from git.

right, so feel free to adjust as you find fit.
The main point was to keep the version information together.

> >  when this commit was merged a880fb1 the dependency on version.cmake was removed:
>
> Did you mean another commit? I don't see any logical change in a880fb1

No, I was simply referring to the fact that in the CMakeLists.txt today it states:
https://github.com/zephyrproject-rtos/zephyr/blob/master/cmake/app/boilerplate.cmake#L150-L158
```
include(${ZEPHYR_BASE}/cmake/version.cmake)  # depends on hex.cmake
....
include(${ZEPHYR_BASE}/cmake/git.cmake)  # depends on version.cmake
```
Before a880fb1, then `git.cmake` was using `KERNEL_VERSION_STRING` defined in `version.cmake`, but with a880fb1 that use was removed, and thus they can now be reordered.

And the comment `# depends on version.cmake` is misleading. 
So it is safe to reorder those to lines in order to have `BUILD_VERSION` available before printing `Zephyr version ....`

>  I'm reluctant to request another round of review from all the people who already approved :-/

i'm fine with taking in this PR as it stands, and then do a follow-up commit.
Just wanted to give a chance to do the update directly in this PR.

Reply:
> Before a880fb1, then git.cmake was using KERNEL_VERSION_STRING defined in version.cmake, but with a880fb1 that use was removed, and thus they can now be reordered.

OK I get it now. When you wrote "use" I misunderstood "some side effect based on it" when it was merely printing it.

> The main point was to keep the version information together.

I agree that would be good and even more so considering I never can remember which one does what or even why there are two of these. I need to read the source / comments every time.

> i'm fine with taking in this PR as it stands, and then do a follow-up commit.

Thank you!